client/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+#![no_std]
+
+use core::fmt::Debug;
+use core::marker::PhantomData;
+use gstd::{msg::MessageFuture, prelude::*, ActorId, MessageId};
+use parity_scale_codec::{Decode, Encode};
+
+#[non_exhaustive]
+#[derive(Default)]
+pub struct SendArgs {
+    value: u128,
+    _sender_id: [u8; 32],
+}
+
+#[non_exhaustive]
+pub struct SendTicket<R: Decode + Debug> {
+    // message we are waiting on
+    f: MessageFuture,
+    // silence the compiler
+    _marker: PhantomData<R>,
+}
+
+impl<R: Decode + Debug> SendTicket<R> {
+    pub async fn result(self) -> Result<R, gstd::errors::Error> {
```

Comment:
1) I would call it `response` in order to avoid any confusion with the `Result` type
2) Rust community suggests using types w/o full path and use aliases if it is needed, for example GStdError